pkg/git/v2/transport.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2020 The Flux CD contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v2
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/sha1"
+	"fmt"
+	"golang.org/x/crypto/ssh"
+	"strings"
+
+	"github.com/fluxcd/source-controller/pkg/git/common"
+	git2go "github.com/libgit2/git2go/v31"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func AuthSecretStrategyForURL(url string) common.AuthSecretStrategy {
+	switch {
+	case strings.HasPrefix(url, "http"):
+		return &BasicAuth{}
+	case strings.HasPrefix(url, "ssh"):
+		return &PublicKeyAuth{}
+	}
+	return nil
+}
+
+type BasicAuth struct{}
+
+func (s *BasicAuth) Method(secret corev1.Secret) (*common.Auth, error) {
+	var username string
+	if d, ok := secret.Data["username"]; ok {
+		username = string(d)
+	}
+	var password string
+	if d, ok := secret.Data["password"]; ok {
+		username = string(d)
```

Comment:
Hi there! I tested this PR and this line here caused some confusion.

I suppose it should be:

```suggestion
		password = string(d)
```

Reply:
My bad good catch!